File: 2-objects-functions-and-types/swap.go
package main

import "fmt"

func swap(a, b *int) {
	ta := *a
	tb := *b
	*a = tb
	*b = ta
}

func main() {
	a := 21
	b := 17

	swap(&a, &b)
	fmt.Printf("main: a = %d, b = %d\n", a, b)
}

File: 2-objects-functions-and-types/swap.c
#include "stdio.h"

void swap(int *a, int *b) {
	int t = *a;
	*a = *b;
	*b = t;
	return;
};

int main(void) {
	int a = 21;
	int b = 17;

	swap(&a, &b);
	printf("main: a = %d, b = %d\n", a, b);
	return 0;
}
